agent/engine/task_manager.go

```diff
@@ -480,6 +523,17 @@ func (mtask *managedTask) isContainerFound(container *api.Container) bool {
 	return found
 }
 
+func (mtask *managedTask) isResourceFound(res taskresource.TaskResource) bool {
+	found := false
+	for _, r := range mtask.Resources {
+		if res == r {
```

Comment:
Will this work, as the `res.knownStatusUnsafe` can be different with the `r.knownStatusUnsafe`, right? The resource may need another method to compare.

Reply:
This check is done before the known status is changed in `handleResourceChange`. 
Nevertheless, comparing the resource names should be sufficient here.